oauth2client/django_util/__init__.py

```diff
@@ -0,0 +1,294 @@
+   def drive_required(request):
+       if request.oauth.has_credentials():
+           service = build(serviceName='drive', version='v2', http=request.oauth.http,
+                developerKey=API_KEY)
+           events = service.files().list().execute()['items']
+           return HttpResponse(str(events))
+       else:
+           return HttpResponse('Here is an OAuth Authorize link: <a href="%s">Authorize</a>' %
+                           request.oauth.get_authorize_redirect())
+
+
+To provide a callback on authorization being completed, use the oauth2_authorized
+signal:
+
+.. code-block:: python
+   :caption: views.py
+   :name: signals
+
+   from oauth2client.django_util.signals import oauth2_authorized
+
+   def test_callback(sender, request, credentials, **kwargs):
+       print "Authorization Signal Received %s" % credentials.id_token['email']
+
+   oauth2_authorized.connect(test_callback)
+
+"""
+
+import httplib2
+
+import django.conf
+from django.core import exceptions
+from django.core import urlresolvers
+from oauth2client import clientsecrets
+from six.moves.urllib import parse
+from oauth2client.django_util import storage
+
+GOOGLE_OAUTH2_DEFAULT_SCOPES = ('email', )
+GOOGLE_OAUTH2_REQUEST_ATTRIBUTE = 'oauth'
+
+
+def _load_client_secrets(filename):
+    """Loads client secrets from the given filename."""
+    client_type, client_info = clientsecrets.loadfile(filename)
+
+    if client_type != clientsecrets.TYPE_WEB:
+        raise ValueError(
+            'The flow specified in {} is not supported, only the WEB flow type is '
+            'supported.'.format(client_type))
+    return client_info['client_id'], client_info['client_secret']
+
+
+def _get_oauth2_client_id_and_secret(settings_instance):
+    """Initializes client id and client secret based on the settings"""
+    if getattr(django.conf.settings, 'GOOGLE_OAUTH2_CLIENT_SECRETS_JSON', None):
+        return _load_client_secrets(
+            settings_instance.GOOGLE_OAUTH2_CLIENT_SECRETS_JSON)
+    elif getattr(settings_instance, "GOOGLE_OAUTH2_CLIENT_ID",
+                 None) and getattr(settings_instance, "GOOGLE_OAUTH2_CLIENT_SECRET", None):
+        return settings_instance.GOOGLE_OAUTH2_CLIENT_ID, \
+            settings_instance.GOOGLE_OAUTH2_CLIENT_SECRET
+    else:
+        raise exceptions.ImproperlyConfigured(
+            "Must specify either GOOGLE_OAUTH2_CLIENT_SECRETS_JSON,or both "
+            "GOOGLE_OAUTH2_CLIENT_ID and GOOGLE_OAUTH2_CLIENT_SECRET in "
+            "settings.py")
+
+
+class OAuth2Settings(object):
+    """Initializes Django OAuth2 Helper Settings
+
+    This class loads the OAuth2 Settings from the Django settings, and then provides
+    those settings as attributes to the rest of the views and decorators in the
+    module.
+
+    Attributes:
+      scopes: A list of OAuth2 scopes that the decorators and views will use as defaults
+      request_prefix: The name of the attribute that the decorators use to attach the UserOAuth2
```

Comment:
I wasn't being careful with column length and wrapping, next commit should be 80 and consistent wrapping.